server/game/gamesteps/conflict/conflictflow.js

```diff
@@ -97,29 +115,69 @@ class ConflictFlow extends BaseStepWithPipeline {
                 }
                 this.conflict.ring = ring;
                 ring.contested = true;
+                this.game.updateCurrentConflict(this.conflict);
                 this.pipeline.queueStep(new InitiateConflictPrompt(this.game, this.conflict, this.conflict.attackingPlayer, false, false, forcedAttackers));
                 return true;
             }
         });
     }
 
+    payAttackerCosts() {
+        this.game.updateCurrentConflict(null);
+        if(!this.conflict.conflictPassed) {
+            const totalFateCost = this.conflict.attackers.reduce((total, card) => total + card.sumEffects(EffectNames.FateCostToAttack), 0);
+            if(!this.conflict.conflictPassed && totalFateCost > 0) {
+                this.game.addMessage('{0} pays {1} fate to declare his attackers', this.conflict.attackingPlayer, totalFateCost);
+                const costEvents = [];
+                Costs.payFate(totalFateCost).addEventsToArray(costEvents, this.game.getFrameworkContext(this.conflict.attackingPlayer));
+                this.game.openEventWindow(costEvents);
+            }
+        }
+    }
+
     initiateConflict() {
         if(this.conflict.conflictPassed) {
             return;
         }
 
-        _.each(this.conflict.attackers, card => card.inConflict = true);
-        this.game.recordConflict(this.conflict);
-    }
+        let provinceSlot = this.conflict.conflictProvince ? this.conflict.conflictProvince.location : Locations.ProvinceOne;
+        let provinceName = (this.conflict.conflictProvince && this.conflict.conflictProvince.facedown) ? provinceSlot : this.conflict.conflictProvince;
+        this.game.addMessage('{0} is initiating a {1} conflict at {2}, contesting {3}', this.conflict.attackingPlayer, this.conflict.conflictType, provinceName, this.conflict.ring);
 
-    payAttackerCosts() {
-        const totalFateCost = this.conflict.attackers.reduce((total, card) => total + card.sumEffects(EffectNames.FateCostToAttack), 0);
-        if(!this.conflict.conflictPassed && totalFateCost > 0) {
-            this.game.addMessage('{0} pays {1} fate to declare his attackers', this.conflict.attackingPlayer, totalFateCost);
-            const costEvents = [];
-            Costs.payFate(totalFateCost).addEventsToArray(costEvents, this.game.getFrameworkContext(this.conflict.attackingPlayer));
-            this.game.openEventWindow(costEvents);
-        }
+        const params = {
+            conflict: this.conflict,
+            type: this.conflict.conflictType,
+            ring: this.conflict.ring,
+            attackers: this.conflict.attackers.slice(),
+            ringFate: this.conflict.ring.fate
+        };
+
+        this.game.raiseEvent(EventNames.OnConflictDeclaredBeforeProvinceReveal, params, event => {
```

Comment:
this should be:
this.game.queueThenEventWindow([this.game.getEvent(EventNames.OnConflictDeclaredBeforeProvinceReveal, params, event => {